src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    let newState;
```

Comment:
The variable `newState` is unnecessary since you can directly assign the new state to `currentState` inside each case, and then push `currentState` to `stateHistory`. This will adhere to the DRY principle and avoid repeating the assignment in each case.